src/tests/GC/Features/BackgroundGC/foregroundgc.cs

```diff
@@ -11,7 +11,8 @@ namespace ForegroundGC
 {
     class ForegroundGC
     {
-        static bool done = false;
+        volatile static bool done = false;
+        static object _lock = new object();
```

Comment:
You shouldn't need to use both `volatile` and a lock.  The lock provides ordering guarantees that make `volatile` redundant.  (`volatile` itself might be enough, but there's no reason to optimize here so that isn't recommended.)

Reply:
Agreed - we were testing both options as viable solutions to the current lack of synchronization with this issue. Let's go with route of the lock and remove `volatile`.